edkrepo/commands/sync_command.py

```diff
@@ -313,23 +311,34 @@ def __update_local_manifest(self, args, config, initial_manifest, workspace_path
                             if source_to_check.remote_url == source.remote_url:
                                 found_source = True
                                 break
+                # If the source that is different came from the old manifest, then it is now outdated and either needs
+                # to be deleted or moved to a archival location.
                 if found_source:
-                    sources_to_remove.append(source)
-                else:
-                    roots = [s.root for s in initial_sources]
+                    roots = [s.root for s in new_sources]
+                    # If there is a source in the new manifest that goes into the same folder name as a source in the
+                    # old manifest, then we need to move that old folder to an archival location.
                     if source.root in roots:
-                        #In theory, this should never happen, we should hit the SYNC_URL_CHANGE error
-                        #this is here as a catch all
                         sources_to_move.append(source)
                     else:
+                        # If it doesn't exist at all in the new manifest, tell the user it is old and no longer used.
+                        sources_to_remove.append(source)
+                else:
+                        # If the source that is different came from the new manifest, then we need to clone that new
+                        # Git repository.
                         sources_to_clone.append(source)
             init_color_console(False)
+            # Move the obsolete Git repositories to archival locations.
             for source in sources_to_move:
                 old_dir = os.path.join(workspace_path, source.root)
                 new_dir = generate_name_for_obsolete_backup(old_dir)
                 print(SYNC_SOURCE_MOVE_WARNING.format(source.root, new_dir))
                 new_dir = os.path.join(workspace_path, new_dir)
-                shutil.move(old_dir, new_dir)
+                try:
+                    shutil.move(old_dir, new_dir)
+                except:
+                    print(SYNC_MOVE_FAILED.format(source.root, new_dir, new_dir, new_dir, source.root))
```

Comment:
With the comment below, this would need to change to:

print(SYNC_MOVE_FAILED.format(old_dir=source.root, new_dir=new_dir))